scripts/serverless-setup.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Create and commit .sam-params-<env> file as SAM deploy parameters
+# The cert ARN and hostedZoneID are not sensitive but requires list all permissions in IAM
+# therefore are created here instead of granting the CI user those permission to fetch on the fly
+# and they should not change often
+
+if [[ "$ENVIRONMENT" == "stage" ]]; then
+  BACKEND_DOMAIN_PREFIX=$stagingBackendSubdomain
+  DOMAIN=$stagingHostRoot
+  FILE_NAME=".sam-params-stage"
+elif [[ "$ENVIRONMENT" == "prod" ]]; then
+  BACKEND_DOMAIN_PREFIX=$productionBackendSubdomain
+  DOMAIN=$productionHostRoot
+  FILE_NAME=".sam-params-prod"
+fi
+
+HOSTED_ZONE_ID=$(aws route53 list-hosted-zones  --query "HostedZones[?Name == '${DOMAIN}.'].Id"  | jq -r '.[0]' | sed 's/\/hostedzone\///')
+GATEWAY_CERTIFICATE_ARN=$(aws acm list-certificates --region ${REGION} | jq -r " .CertificateSummaryList [] | select( .DomainName == \"${BACKEND_DOMAIN_PREFIX}${DOMAIN}\") | .CertificateArn")
```

Comment:
You can also do this with just the AWS cli tool: `aws acm list-certificates --query "CertificateSummaryList[?DomainName=='mydomain.com'].CertificateArn | [0]" --output text`
It will return with a non-zero response code if a matching domain isn't found.